compiler/rustc_public/src/lib.rs

```diff
@@ -300,3 +295,26 @@ impl rustc_public_bridge::bridge::Allocation<compiler_interface::BridgeTys>
         }
     }
 }
+
+#[derive(Clone, Copy, Hash, PartialEq, Eq, Default)]
+#[derive(serde::Serialize)] // TODO: Don't.
```

Comment:
This should indeed not be serializable and it is an implementation detail due to the lack of negative traits.

Reply:
Done in 97e6df4. It's sadly not as simple as using `#[serde(skip)]`, as that still moves serde from serializing just the single field, to a list containing the single field: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=b4aafe4ff5312c4ddedd5e17ee361d5a

Is there somewhere that tests for this could go?